content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md

```diff
@@ -0,0 +1,410 @@
+  - /influxdb/cloud-dedicated/reference/client-libraries/v3/
+---
+
+## Troubleshoot query performance
+
+Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:
+
+<!-- TOC -->
+
+- [Troubleshoot query performance](#troubleshoot-query-performance)
+  - [EXPLAIN and ANALYZE](#explain-and-analyze)
+  - [Enable trace logging](#enable-trace-logging)
+    - [Extract a trace ID](#extract-a-trace-id)
+  - [Retrieve system information](#retrieve-system-information)
+    - [Retrieve query instance information](#retrieve-query-instance-information)
+
+<!-- /TOC -->
+
+### EXPLAIN and ANALYZE
+
+To view the query engine's execution plan and metrics for an SQL query, prepend [`EXPLAIN`](/influxdb/cloud-dedicated/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-dedicated/reference/sql/explain/#explain-analyze) to the query.
+The report can reveal query bottlenecks such as a large number of table scans or parquet files, and can help triage the question, "Is the query slow due to the amount of work required or due to a problem with the schema, compactor, etc.?"
+
+The following example shows how to use the InfluxDB v3 Python client library and pandas to view `EXPLAIN` and `EXPLAIN ANALYZE` results for a query:
+
+<!-- Import for tests and hide from users.
+```python
+import os
+```
+-->
+<!--pytest-codeblocks:cont-->
+{{% code-placeholders "DATABASE_(NAME|TOKEN)|APP_REQUEST_ID" %}}
+```python
+from influxdb_client_3 import InfluxDBClient3
+import pandas as pd
+import tabulate # Required for pandas.to_markdown()
+
+def explain_and_analyze():
+  print('Use SQL EXPLAIN and ANALYZE to view query plan information.')
+
+  # Instantiate an InfluxDB client.
+  client = InfluxDBClient3(token = f"DATABASE_TOKEN",
+                          host = f"{{< influxdb/host >}}",
+                          database = f"DATABASE_NAME")
+
+  sql_explain = '''EXPLAIN SELECT *
+        FROM home
+        WHERE time >= now() - INTERVAL '90 days'
+        ORDER BY time'''
+
+  table = client.query(sql_explain)
+  df = table.to_pandas()
+
+  sql_explain_analyze = '''EXPLAIN ANALYZE SELECT *
+      FROM home
+      WHERE time >= now() - INTERVAL '90 days'
+      ORDER BY time'''
+
+  table = client.query(sql_explain_analyze)
+
+  # Combine the Dataframes and output the plan information.
+  df = pd.concat([df, table.to_pandas()])
+
+  assert df.shape == (3, 2) and df.columns.to_list() == ['plan_type', 'plan']
+  print(df[['plan_type', 'plan']].to_markdown(index=False))
+
+  client.close()
+
+explain_and_analyze()
+```
+{{% /code-placeholders %}}
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} database
+- {{% code-placeholder-key %}}`DATABASE_TOKEN`{{% /code-placeholder-key %}}: a [database token](/influxdb/cloud-dedicated/admin/tokens/) with sufficient permissions to the specified database
+
+The output is similar to the following:
```

Comment:
There may be differences in the output between SQL and InfluxQL. Maybe some tabs here?